### PR TITLE
Yet more `rustc_mir_dataflow` cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
A few more cleanups.

r? @cjgillot

### CODE REVIEW

File: compiler/rustc_mir_dataflow/src/framework/direction.rs [hunk 1]
```diff
@@ -12,6 +12,16 @@ pub trait Direction {
 
     const IS_BACKWARD: bool = !Self::IS_FORWARD;
 
+    fn apply_effects_in_block_and_join_state_into_successors_of<'mir, 'tcx, A>(
```
Comment (cjgillot) [inline compiler/rustc_mir_dataflow/src/framework/direction.rs:15]: Could you shorten the name?

**Comment by cjgillot** [comment]
Thanks. r=me with a naming nit

File: compiler/rustc_mir_dataflow/src/framework/direction.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_mir_dataflow/src/framework/direction.rs:15]: Sure. I'll just use `apply_effects_in_block`, I think that's good enough.

**Comment by nnethercote** [comment]
I updated the name.

@bors r=cjgillot

**Pull request merged**